salt/beacons/__init__.py

```diff
@@ -95,6 +97,17 @@ def process(self, config, grains):
                         continue
                 # Update __grains__ on the beacon
                 self.beacons[fun_str].__globals__['__grains__'] = grains
+
+                # Run the validate function if it's available,
+                # otherwise there is a warning about it being missing
+                if validate_str in self.beacons:
+                    valid, vcomment = self.beacons[validate_str](b_config[mod])
+
+                    if not valid:
+                        log.info('Beacon {0} configuration invalid, '
```

Comment:
Now I know the whole story behind validate, thanks! It makes sense now.

About logging, yes I'm suggesting you use string replacement on log calls.
We use string formatting all over and we should keep using it, allows for improved readability with a small performance penalty that we decided to accept. Using them on log calls just becomes a double penalty :)

I'm not saying replace all string formatting on all log calls, but if you add or update a log call, switch it to string replacement.